lib/Dialect/TTIR/Analysis/LegalGridAnalysis.cpp

```diff
@@ -37,6 +37,10 @@ layout_get_shard_shape(const mlir::tt::LayoutAttr &layout) {
   return shard_shape;
 }
 
+std::string layout_get_tensor_layout(const mlir::tt::LayoutAttr &layout) {
```

Comment:
One way to get rid of these magic strings and rely to definitions at `TTOpsEnums.td`

```
def TT_OperandConstraintScalar : I32BitEnumAttrCaseBit<"Scalar", 3, "scalar">;
def TT_OperandConstraintTile : I32BitEnumAttrCaseBit<"Tile", 4, "tile">;
```

would be to add `let stringToSymbolFnName = "OperandConstraintStringToEnum";` at `def TT_OperandConstraint` and than you can use `mlir::tt::stringifyOperandConstraint(mlir::tt::OperandConstraint::Scalar)`
and `mlir::tt::stringifyOperandConstraint(mlir::tt::OperandConstraint::Tile)` instead of magic strings

there might be some connection of OperandConstraint with layout, but I dont see it at the moment.